pkg/controller/cloud/node_controller.go

```diff
@@ -239,9 +239,39 @@ func (cnc *CloudNodeController) MonitorNode() {
 		// from the cloud provider. If node cannot be found in cloudprovider, then delete the node immediately
 		if currentReadyCondition != nil {
 			if currentReadyCondition.Status != v1.ConditionTrue {
+				// we need to check this first to get taint working in similar in all cloudproviders
+				// current problem is that shutdown nodes are not working in similar way ie. all cloudproviders
+				// does not delete node from kubernetes cluster when instance it is shutdown see issue #46442
+				exists, err := instances.InstanceStoppedByProviderID(node.Spec.ProviderID)
+				if err != nil {
+					glog.Errorf("Error getting data for node %s from cloud: %v", node.Name, err)
+					continue
+				}
+
+				if exists {
+					// if node is shutdown add shutdown taint
+					cloudTaint := getTaint(node.Spec.Taints, algorithm.TaintNodeStopped)
+					if cloudTaint == nil {
+						taint := v1.Taint{
+							Key:    algorithm.TaintNodeStopped,
+							Value:  "true",
+							Effect: v1.TaintEffectNoSchedule,
+						}
+						glog.V(4).Infof("Appending stopped taint to node: %s", node.Name)
+						node.Spec.Taints = append(node.Spec.Taints, taint)
+
+						_, err = cnc.kubeClient.CoreV1().Nodes().Update(node)
```

Comment:
Is this an update or a patch ?

Reply:
patch would be better, but then I must implement more code. For instance https://github.com/zetaab/kubernetes/blob/master/pkg/controller/cloud/node_controller.go#L366 is using update as well in this file.

Reply:
There's #56350 to fix this, also there was some constrains that made us use update. If you use update you might overwrite Status, which is something we don't want.